### PR TITLE
Feat: add ICQ configuration NTRN-248

### PR DESCRIPTION
**Pull request by foxpy**
This PR adds template ICQ configuration and docs on how to fill it.

Merge AFTER https://github.com/neutron-org/neutron-query-relayer/pull/51

### CODE REVIEW

File: neutron-devnet-1/icq-relayer/.env [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+# Leave here RPC and REST addresses of your node
+RELAYER_NEUTRON_CHAIN_RPC_ADDR=tcp://127.0.0.1:26657
+RELAYER_NEUTRON_CHAIN_REST_ADDR=http://127.0.0.1:1317
+
+# Chain ID of Quark tetsnet
+RELAYER_NEUTRON_CHAIN_CHAIN_ID=quark-1
+
+# Path to home directory where yours keys are stored
+RELAYER_NEUTRON_CHAIN_HOME_DIR=/path/to/home
```
Comment (albertandrejev) [inline neutron-devnet-1/icq-relayer/.env:12]: Please leave these option empty (I guess in this case relayer will halt?)

File: neutron-devnet-1/icq-relayer/.env [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# If you don't do that, relayer will submit TX results for all ICQs on the network and drain its money quickly
+RELAYER_REGISTRY_ADDRESSES=
+
+RELAYER_ALLOW_TX_QUERIES=true
+RELAYER_ALLOW_KV_CALLBACKS=true
+RELAYER_MIN_KV_UPDATE_PERIOD=1
+
+# Path to database storage for ICQ relayer
+# You may specify empty directory and ICQ relayer will populate it automatically
+RELAYER_STORAGE_PATH=/path/to/storage
```
Comment (albertandrejev) [inline neutron-devnet-1/icq-relayer/.env:65]: Please leave these option empty (I guess in this case relayer will halt?)

File: neutron-devnet-1/icq-relayer/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Configuration for Neutron Query Relayer
+
+## Install relayer
```
Comment (albertandrejev) [inline neutron-devnet-1/icq-relayer/README.md:3]: Please put required minimum Go version here.

**Review by albertandrejev — CHANGES_REQUESTED** [review]

File: neutron-devnet-1/icq-relayer/.env (hunk 1 above)
Comment (foxpy) [reply neutron-devnet-1/icq-relayer/.env:12]: Tested, it doesn't halt. It fallbacks to "keyring-test" somewhere inside.

File: neutron-devnet-1/icq-relayer/.env (hunk 2 above)
Comment (foxpy) [reply neutron-devnet-1/icq-relayer/.env:65]: Yes, it will. I will change this line.

File: neutron-devnet-1/icq-relayer/README.md (hunk 3 above)
Comment (foxpy) [reply neutron-devnet-1/icq-relayer/README.md]: Fixed

**Review by albertandrejev — DISMISSED** [review]

File: quark/icq-relayer/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+- RELAYER_NEUTRON_CHAIN_RPC_ADDR
+- RELAYER_NEUTRON_CHAIN_REST_ADDR
+- RELAYER_NEUTRON_CHAIN_HOME_DIR
+- RELAYER_NEUTRON_CHAIN_CONNECTION_ID
+- RELAYER_NEUTRON_CHAIN_CLIENT_ID
+- RELAYER_TARGET_CHAIN_RPC_ADDR
+- RELAYER_TARGET_CHAIN_CHAIN_ID
+- RELAYER_TARGET_CHAIN_ACCOUNT_PREFIX
+- RELAYER_TARGET_CHAIN_VALIDATOR_ACCOUNT_PREFIX
+- RELAYER_TARGET_CHAIN_CONNECTION_ID
```
Comment (NeverHappened) [inline quark/icq-relayer/README.md:38]: These variables won't be needed to fill in if we merge this https://github.com/neutron-org/neutron-query-relayer/pull/51
`RELAYER_NEUTRON_CHAIN_CLIENT_ID` , `RELAYER_TARGET_CHAIN_CONNECTION_ID`, `RELAYER_TARGET_CHAIN_CLIENT_ID`

File: quark/icq-relayer/.env [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+RELAYER_NEUTRON_CHAIN_KEYRING_BACKEND=test
+RELAYER_NEUTRON_CHAIN_OUTPUT_FORMAT=json
+RELAYER_NEUTRON_CHAIN_SIGN_MODE_STR=direct
+
+
+# Leave here RPC address of Cosmos/Juno testnet node
+RELAYER_TARGET_CHAIN_RPC_ADDR=
+
+# Chain ID of Cosmos/Juno testnet
+RELAYER_TARGET_CHAIN_CHAIN_ID=
```
Comment (NeverHappened) [inline quark/icq-relayer/.env:40]: see https://github.com/neutron-org/testnets/pull/108/files#r1015570016

File: quark/icq-relayer/.env [hunk 6]
```diff
@@ -0,0 +1,68 @@
+# This is a configuration template for ICQ relayer config for Quark testnet
+
+
+# Leave here RPC and REST addresses of your node
+RELAYER_NEUTRON_CHAIN_RPC_ADDR=tcp://127.0.0.1:26657
+RELAYER_NEUTRON_CHAIN_REST_ADDR=http://127.0.0.1:1317
+
+# Chain ID of Quark tetsnet
+RELAYER_NEUTRON_CHAIN_CHAIN_ID=quark-1
```
Comment (NeverHappened) [inline quark/icq-relayer/.env:9]: also do not need if [this](https://github.com/neutron-org/neutron-query-relayer/pull/51/files) gets merged

File: quark/icq-relayer/.env [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+# Increase gas prices in case if relayer cannot submit
+RELAYER_NEUTRON_CHAIN_GAS_PRICES=0.05untrn
+RELAYER_NEUTRON_CHAIN_GAS_LIMIT=10000000
+RELAYER_NEUTRON_CHAIN_GAS_ADJUSTMENT=1.4
+
+# Connection ID on Quark testnet for Cosmos/Juno testnet
+RELAYER_NEUTRON_CHAIN_CONNECTION_ID=connection-0
+
+# Client ID associated with connection ID on Quark testnet for Cosmos/Juno testnet
+RELAYER_NEUTRON_CHAIN_CLIENT_ID=07-tendermint-0
```
Comment (NeverHappened) [inline quark/icq-relayer/.env:28]: also do not need if [this](https://github.com/neutron-org/neutron-query-relayer/pull/51/files) gets merged

File: quark/icq-relayer/.env [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+# Chain ID of Quark tetsnet
+RELAYER_NEUTRON_CHAIN_CHAIN_ID=quark-1
+
+# Path to home directory where yours keys are stored
+RELAYER_NEUTRON_CHAIN_HOME_DIR=/path/to/home
+
+# Name of signing key for ICQ relayer, ideally you should allocate an exclusive account for relayer's needs
+RELAYER_NEUTRON_CHAIN_SIGN_KEY_NAME=my_wallet
+
+RELAYER_NEUTRON_CHAIN_TIMEOUT=1000s
```
Comment (NeverHappened) [inline quark/icq-relayer/.env:17]: don't need to specify (has default value) if this gets merged https://github.com/neutron-org/neutron-query-relayer/pull/51/files

File: quark/icq-relayer/.env [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+# Leave here RPC address of Cosmos/Juno testnet node
+RELAYER_TARGET_CHAIN_RPC_ADDR=
+
+# Chain ID of Cosmos/Juno testnet
+RELAYER_TARGET_CHAIN_CHAIN_ID=
+
+RELAYER_TARGET_CHAIN_ACCOUNT_PREFIX=juno # or cosmos
+RELAYER_TARGET_CHAIN_VALIDATOR_ACCOUNT_PREFIX=junovaloper # or cosmosvaloper
+RELAYER_TARGET_CHAIN_TIMEOUT=1000s
```
Comment (NeverHappened) [inline quark/icq-relayer/.env:44]: don't need to specify (has default value) if this gets merged https://github.com/neutron-org/neutron-query-relayer/pull/51/files

**Review by NeverHappened — CHANGES_REQUESTED** [review]
as we decided to merge https://github.com/neutron-org/neutron-query-relayer/pull/51/files
we need to change this according to new simplified config

File: quark/icq-relayer/README.md (hunk 4 above)
Comment (foxpy) [reply quark/icq-relayer/README.md:38]: Please ping me when it is merged, and I will be happy to remove them!

Comment (foxpy) [reply quark/icq-relayer/README.md:38]: Oh, wait, so are we 100% certain to merge it? Then I have to remove these options right now.

Comment (NeverHappened) [reply quark/icq-relayer/README.md:38]: @oopcode are we gonna merge it?

**Review by albertandrejev — DISMISSED** [review]

**Review by pr0n00gler — APPROVED** [review]

File: quark/icq-relayer/.env [hunk 10]
```diff
@@ -36,19 +28,6 @@ RELAYER_NEUTRON_CHAIN_SIGN_MODE_STR=direct
 # Leave here RPC address of Cosmos/Juno testnet node
 RELAYER_TARGET_CHAIN_RPC_ADDR=
 
-# Chain ID of Cosmos/Juno testnet
-RELAYER_TARGET_CHAIN_CHAIN_ID=
-
-RELAYER_TARGET_CHAIN_ACCOUNT_PREFIX=juno # or cosmos
-RELAYER_TARGET_CHAIN_VALIDATOR_ACCOUNT_PREFIX=junovaloper # or cosmosvaloper
```
Comment (NeverHappened) [inline quark/icq-relayer/.env]: these two options were necessary?

Comment (NeverHappened) [reply quark/icq-relayer/.env]: fixed it

**Review by NeverHappened — APPROVED** [review]

**Pull request merged**